pkg/llms/validate/validator.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023 KubeAGI.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"fmt"
+	arcadiav1alpha1 "github.com/kubeagi/arcadia/api/v1alpha1"
+	"github.com/kubeagi/arcadia/pkg/llms"
+	"github.com/kubeagi/arcadia/pkg/llms/zhipuai"
+	"net/http"
+)
+
+type Validator interface {
+	Validate(auth string, instance *arcadiav1alpha1.LLM) error
+}
+
+type OpenAIValidator struct {
+	response llms.Response
+}
+
+func (v *OpenAIValidator) Validate(auth string, instance *arcadiav1alpha1.LLM) error {
+	// Validate OpenAI type CRD LLM Instance
+	// instance.Spec.URL should be like "https://api.openai.com/"
+
+	if auth == "" {
+		// TODO: maybe we should consider local pseudo-openAI LLM worker that doesn't require an apiKey?
+		return fmt.Errorf("auth is empty")
+	}
+
+	testAuth := "Bearer " + auth // openAI official requirement
+	testURL := instance.Spec.URL + "/v1/models"
+
+	req, err := http.NewRequest("GET", testURL, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Authorization", testAuth)
+	req.Header.Set("Content-Type", "application/json")
+
+	cli := &http.Client{}
+	resp, err := cli.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("returns unexpected status code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+type ZhiPuAIValidator struct {
```

Comment:
Do not need to define such a struct. You can implemt this validation in pkg/llms/zhipuai